libtelemetry/src/main/java/com/mapbox/android/telemetry/metrics/TelemetryMetricsClient.java

```diff
@@ -0,0 +1,54 @@
+package com.mapbox.android.telemetry.metrics;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+
+import java.util.concurrent.TimeUnit;
+
+public class TelemetryMetricsClient {
+  private static final String TELEMETRY_METRICS_USER_AGENT = "mapbox-android-metrics";
+  private static TelemetryMetricsClient telemetryMetricsClient;
+  private static final Object lock = new Object();
+  private final TelemetryMetrics telemetryMetrics;
+
+  @VisibleForTesting
+  TelemetryMetricsClient(@NonNull TelemetryMetrics telemetryMetrics) {
+    this.telemetryMetrics = telemetryMetrics;
+  }
+
+  public static TelemetryMetricsClient install(@NonNull Context context) {
+    Context applicationContext;
+    if (context.getApplicationContext() == null) {
+      // In shared processes content providers getApplicationContext() can return null.
+      applicationContext = context;
+    } else {
+      applicationContext = context.getApplicationContext();
+    }
+
+    // TODO: fetch metrics state file
+    TelemetryMetrics metrics = new TelemetryMetrics(TimeUnit.HOURS.toMillis(24));
+    synchronized (lock) {
+      if (telemetryMetricsClient == null) {
+        telemetryMetricsClient = new TelemetryMetricsClient(metrics);
+      }
+    }
+    return telemetryMetricsClient;
+  }
+
+  @NonNull
+  public static TelemetryMetricsClient getInstance() {
+    synchronized (lock) {
```

Comment:
yes, lock isn't thread safe and can be accessed via `getInstance()` method on another thread. this is a future proof just in case this class is instantiated differently.